privacyidea_pam.py

```diff
@@ -122,6 +169,45 @@ def authenticate(self, password):
 
         return rval
 
+    def challenge_response(self, transaction_id, challenge):
+        rval = self.pamh.PAM_SYSTEM_ERR
+
+        message = self.pamh.Message(self.pamh.PAM_PROMPT_ECHO_OFF, challenge)
+        response = self.pamh.conversation(message)
+        chal_response = json.loads(response.resp)
```

Comment:
I tried to run it but I break at this point. The ssh-u2f.py client seems to return a wrong response object from the conversation.

response.resp == "u2f".

<pre>
Mar  3 18:27:43 puckel sshd: <Response_type object at 0x7f47188e7650>
Mar  3 18:27:43 puckel sshd: u2f
Mar  3 18:27:43 puckel sshd: Traceback (most recent call last):
#012  File "/home/cornelius/src/pam_python_u2f/privacyidea_pam.py", line 236, in pam_sm_authenticate
#012    rval = Auth.authenticate(pamh.authtok)
#012  File "/home/cornelius/src/pam_python_u2f/privacyidea_pam.py", line 157, in authenticate
#012    challenge)
#012  File "/home/cornelius/src/pam_python_u2f/privacyidea_pam.py", line 181, in challenge_response
#012    chal_response = json.loads(response.resp)
#012  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
#012    return _default_decoder.decode(s)
#012  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
#012    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
#012  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
#012    raise ValueError("No JSON object could be decoded")
#012ValueError: No JSON object could be decoded
</pre>